src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs

```diff
@@ -207,7 +207,11 @@ typeOfSetupValue' cc env nameEnv val =
                  case memTy' of
                    Crucible.ArrayType n memTy''
                      | fromIntegral i < n -> return (Crucible.PtrType (Crucible.MemType memTy''))
-                     | otherwise -> fail $ "typeOfSetupValue: array type index out of bounds: " ++ show (i, n)
+                     | otherwise -> fail $ unlines $
+                         [ "typeOfSetupValue: Array type index out of bounds: "
+                         , "  Index:        " ++ show i
+                         , "  Array length: " ++ show n
+                         ]
```

Comment:
It might be good to have a general discussion and policy decision on this topic: my preference is to keep these things on a single line where feasible to facilitate subsequent analysis.

Examples are running "grep", "ag", whatever on the output and especially when performing additional processing on these (e.g. `runtests.sh | grep 'out of bounds' | sort -u`)

The tension here is in providing nice output for user consumption.

Reply:
I have no preference for one line vs many, my central concern here is giving names to these numbers.